### PR TITLE
Migrate to the UniswapV2 oracle

### LINKED ISSUE #6314: Upgrade to Uniswap v2
https://twitter.com/UniswapExchange/status/1242183677590810626

We are most interested in the new price oracle feature.
Need to confirm they gain liquidity before we make the swap.

### PR DESCRIPTION
**Pull request by NickCuso**
# Description

# Issues

Fixes #6314
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/Unlock.sol [hunk 1]
```diff
@@ -203,14 +217,11 @@ contract Unlock is
...
       address tokenAddress = IPublicLock(msg.sender).tokenAddress();
-      if(tokenAddress != address(0)) {
-        // If priced in an ERC-20 token, find the supported uniswap exchange
-        IUniswapExchange exchange = uniswapExchanges[tokenAddress];
-        if(address(exchange) != address(0)) {
-          valueInETH = exchange.getTokenToEthInputPrice(_value);
-        } else {
-          // If the token type is not supported, assume 0 value
-          valueInETH = 0;
+      if(tokenAddress != address(0) && tokenAddress != weth) {
```
Comment (julien51) [inline smart-contracts/contracts/Unlock.sol:220]: Does that mean we will nto account for locks in Weth? I think we should also use them?

Comment (NickCuso) [reply smart-contracts/contracts/Unlock.sol:220]: UniswapV2 doesn't support ETH under the hood. So we are converting to weth, and if the tokenAddress is ETH or WETH then we skip Uniswap and use the value directly in the else block below.

Comment (julien51) [reply smart-contracts/contracts/Unlock.sol:220]: I think in that case we should just set `valueInETH = _value`

Comment (NickCuso) [reply smart-contracts/contracts/Unlock.sol:220]: Correct, that's what we do in the else block below

**Pull request merged**